src/devices/machine/ldp1450hle.cpp

```diff
@@ -627,6 +896,35 @@ void sony_ldp1450hle_device::update_video_enable()
 }
 
 
+bitmap_yuy16 sony_ldp1450hle_device::osd_char_gen(uint8_t idx)
+{
+	uint16_t white = 0xeb80;
+	uint16_t black = 0x0080;
+
+	// iterate over pixels
+	const u16 *chdataptr = &text_bitmap[idx][0];
+
+	bitmap_yuy16 char_bmp = bitmap_yuy16(16,16);
+	for (u8 y = 0; y < 16; y++)
+	{
+		u16 chdata = *chdataptr++;
+
+		for (u8 x = 0; x < 16; x++, chdata >>= 1)
+		{
+			if (chdata & 0x01)
+			{
+				char_bmp.pix(y, x) = white;
+			}
+			else
+			{
+				char_bmp.pix(y, x) = black;
+			}
+		}
+	}
+	return char_bmp;
+}
+
```

Comment:
This kind of on-the-fly generation looks inefficient, but I probably need to play with the driver at hand to find a better way to do it (for example it may be just a single texture that scissors the necessary content instead of instantiating 96 different `bitmap_yuy16` (sic)).

Reply:
Yeah, I didn't particularly care for it, but it's something I'd like feedback on. The whole draw texture to scale and overlay probably needs a look at. I'll fix the other issues you mention at least, so there's a better framework to work on.

Reply:
Is this overlay stuff documented anywhere?

Reply:
There is a programming manual, that sort of explains it:
https://www.dragons-lair-project.com/tech/manuals/[Sony_LDM-G1000_Protocol_and_Command_Manual_for_Sony_LDP_Series_Videodisc_Players.pdf](https://www.dragons-lair-project.com/tech/manuals/Sony_LDM-G1000_Protocol_and_Command_Manual_for_Sony_LDP_Series_Videodisc_Players.pdf)
Page 44 of the PDF talks through the controls, and the appendix in 62 shows the charset, such as it is.

Reply:
Furthermore, Matt Ownby did some probing of real hardware to see if there was a simpler approach, for completeness I'll link to that blog post. I got a bit more out of that than the manual to be fair
https://my-cool-projects.blogspot.com/2013/04/sony-ldp-1450-text-overlay-horizontal.html

Reply:
There's a whole lot of tech debt on the LD department, not confined to this core alone. Been pondering from a while if we are ready for real RS-232C comms with a computer, manual mentions SMC-70 and SMC-3000/DOS-compatible capable at very least.

I'll probably merge this as-is when I'm ready and willing to do stuff there.

Reply:
Well, my experience with this was with the PAL version and an MSX over the native basic, surprisingly versatile. If you don't need any more from this end then I'm happy to hand this off.